src/tm_devices/drivers/pi/pi_device.py

```diff
@@ -633,6 +706,70 @@ def set_and_check(
             check = ""
         return check
 
+    def set_if_needed(  # noqa: PLR0913
+        self,
+        command: str,
+        value: Union[str, float],
+        tolerance: float = 0,
+        percentage: bool = False,
+        remove_quotes: bool = False,
+        custom_message_prefix: str = "",
+        *,
+        expected_value: Optional[Union[str, float]] = None,
+        opc: bool = False,
+        allow_empty: bool = False,
+        verify_value: bool = False,
+    ) -> Tuple[bool, str]:
+        """Query the command's field and update it if the value does not match the input.
+
+        Args:
+            command: The command to send.
+                For example: ``:AFG:FUNCTION``
+            value: The value being set by the command.
+                For example: ``SINE``
+            tolerance: The acceptable difference between two floating point values.
+            percentage: A boolean indicating what kind of tolerance check to perform.
+                 False means absolute tolerance: +/- tolerance.
+                 True means percent tolerance: +/- (tolerance / 100) * value.
+            remove_quotes: Set this to True to remove all double quotes from the returned value.
+            custom_message_prefix: A custom message to be prepended to the failure message.
+            expected_value: An optional, alternative value expected to be returned.
+            opc: Boolean indicating if ``*OPC?`` should be queried after sending the command.
+            allow_empty: Set this to True if an empty return string is permitted.
+            verify_value: Boolean indicating to verify value after write.
+
+        Returns:
+            Tuple containing the boolean value indicating if the command needed to be set and
+            the value returned from the query.
+        """
+        try:
+            query_passed, actual_value = self.query_response(
```

Comment:
Potential nit but instead of using a try catch with query response would it make sense to just query the value instead and compare it to the actual value

Reply:
Query response does this inherently, I believe it makes more sense to just utilize already existing functionality